File: Segundo/Segundo_Cuatrimestre/Informatica/PDOO/SUBIR/MARCIANITOS_P3/Marcianitos/src/deepspace/Loot.java
package deepspace;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author mario
 */
class Loot {
    
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
	
	private boolean getEfficient;
	private boolean spaceCity;
    
    Loot(int supplies, int weapons, int shields, int hangars, int medals, boolean ef, boolean city){
        
        nSupplies=supplies;
        nWeapons=weapons;
        nShields=shields;
        nHangars= hangars;
        nMedals=medals;
		
		getEfficient = ef;
		spaceCity = city;
    }
    
    LootToUI getUIversion() {
        return new LootToUI(this);
    }
    
    public int getNSupplies(){
        
        return nSupplies;
    }
    
    public int getNWeapons(){
        
        return nWeapons;
    }
    
    public int getNShields(){
        
        return nShields;
    }
    
    public int getNHangars(){
        
        return nHangars;
    }
     
    public int getNMedals(){
        
        return nMedals;
    }
	
	public boolean getEfficient (){
		return (getEfficient);
	}
	
	public boolean spaceCity (){
		return (spaceCity);
	}
    
    public String toString(){
        
        String salida;
        
        salida = "Supplies=" + nSupplies + ", Weapons=" + nWeapons + ", Shields=" + nShields + ", Hangars=" + nHangars;

        return salida;
    }
    
}

File: Segundo/Segundo_Cuatrimestre/Informatica/PDOO/SUBIR/MARCIANITOS_P3/RUBY/src/Loot.rb
#encoding: UTF-8
require_relative 'LootToUI'

module Deepspace
	class Loot
		def initialize (supplies,weapons,shields,hangars,medals)  # son todos int
			@nSupplies = supplies
			@nWeapons = weapons
			@nShields = shields
			@nHangars = hangars
			@nMedals = medals
		end

		attr_reader :nSupplies
		
		attr_reader :nWeapons

		attr_reader :nShields
	
		attr_reader :nHangars

		attr_reader :nMedals
		
		def getUIversion()
			return LootToUI.new(self)
		end
		
		def to_s
			return "Supplies: #{@nSupplies}, Weapons: #{@nWeapons}, Shields: #{@nShields}, Hangars: #{@nHangars}, Medals: #{@nMedals}"
		end
	end

end
